substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -487,10 +490,15 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
              */
             Node singleFunctionality = null;
             ReturnNode singleReturn = null;
+            ValueNode singleNewInstance = null;
             for (Node node : graph.getNodes()) {
                 if (node == graph.start() || node instanceof ParameterNode || node instanceof ConstantNode || node instanceof FrameState) {
                     /* Ignore the allowed framework around the nodes we care about. */
-                } else if (node instanceof MethodCallTargetNode) {
+                } else if (node instanceof DynamicNewInstanceNode && singleNewInstance == null && ((DynamicNewInstanceNode) node).getInstanceType().isConstant()) {
+                    singleNewInstance = (DynamicNewInstanceNode) node;
+                } else if (node instanceof NewInstanceNode && singleNewInstance == null) {
+                    singleNewInstance = (NewInstanceNode) node;
+                } else if (node instanceof MethodCallTargetNode || node instanceof InstanceOfNode || node instanceof FixedGuardNode) {
```

Comment:
You cannot just ignore `InstanceOfNode` and `FixedGuardNode`. That would mean you omit necessary type checks.

Reply:
This should be fixed.  Removed these and added a ClassCast check in the transplanted nodes.